### PR TITLE
implements controller, created new DTOs, and DTOs' Mappers

### PR DESCRIPTION
**Pull request by Daniel1004K**

### CODE REVIEW

**Review by Veronika-Alekseienko — APPROVED** [review]
Well done!)

File: src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java [hunk 1]
```diff
@@ -24,6 +25,28 @@ public MovieSessionDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
+    @Override
+    public MovieSession add(MovieSession movieSession) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java:29]: isn't it already implemented in AbstractDao?

File: src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -53,4 +76,47 @@ public MovieSessionDaoImpl(SessionFactory sessionFactory) {
...
+    }
+
+    @Override
+    public void delete(Long id) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(id);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java:109]: does it work?

File: src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -53,4 +76,47 @@ public MovieSessionDaoImpl(SessionFactory sessionFactory) {
...
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't delete movie session ", e);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java:115]: add id to message please

File: src/main/java/mate/academy/spring/service/mapper/CinemaHallDtoMapper.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package mate.academy.spring.service.mapper;
+
+import mate.academy.spring.model.CinemaHall;
+import mate.academy.spring.model.dto.request.CinemaHallRequestDto;
+import mate.academy.spring.model.dto.response.CinemaHallResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CinemaHallDtoMapper {
+    public CinemaHallResponseDto parse(CinemaHall cinemaHall) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/service/mapper/CinemaHallDtoMapper.java:10]:
```suggestion
    public CinemaHallResponseDto toDto(CinemaHall cinemaHall) {
```

File: src/main/java/mate/academy/spring/service/mapper/MovieDtoMapper.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package mate.academy.spring.service.mapper;
+
+import mate.academy.spring.model.Movie;
+import mate.academy.spring.model.dto.request.MovieRequestDto;
+import mate.academy.spring.model.dto.response.MovieResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieDtoMapper {
+    public MovieResponseDto parse(Movie movie) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/service/mapper/MovieDtoMapper.java:10]: same

File: src/main/java/mate/academy/spring/service/mapper/MovieSessionDtoMapper.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    private CinemaHallService cinemaHallService;
+
+    @Autowired
+    public MovieSessionDtoMapper(MovieService movieService,
+            CinemaHallService cinemaHallService) {
+        this.movieService = movieService;
+        this.cinemaHallService = cinemaHallService;
+    }
+
+    public MovieSessionResponseDto parse(MovieSession movieSession) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/service/mapper/MovieSessionDtoMapper.java:25]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**